matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/filter/SaveFilterTask.kt

```diff
@@ -26,44 +25,32 @@ import javax.inject.Inject
 /**
  * Save a filter, in db and if any changes, upload to the server.
  */
-internal interface SaveFilterTask : Task<SaveFilterTask.Params, Unit> {
+internal interface SaveFilterTask : Task<SaveFilterTask.Params, String> {
 
     data class Params(
-            val filterPreset: FilterService.FilterPreset
+            val filter: Filter
     )
 }
 
 internal class DefaultSaveFilterTask @Inject constructor(
         @UserId private val userId: String,
         private val filterAPI: FilterApi,
         private val filterRepository: FilterRepository,
-        private val globalErrorReceiver: GlobalErrorReceiver
+        private val globalErrorReceiver: GlobalErrorReceiver,
 ) : SaveFilterTask {
 
-    override suspend fun execute(params: SaveFilterTask.Params) {
-        val filterBody = when (params.filterPreset) {
-            FilterService.FilterPreset.ElementFilter -> {
-                FilterFactory.createElementFilter()
-            }
-            FilterService.FilterPreset.NoFilter -> {
-                FilterFactory.createDefaultFilter()
-            }
-        }
-        val roomFilter = when (params.filterPreset) {
-            FilterService.FilterPreset.ElementFilter -> {
-                FilterFactory.createElementRoomFilter()
-            }
-            FilterService.FilterPreset.NoFilter -> {
-                FilterFactory.createDefaultRoomFilter()
-            }
-        }
-        val updated = filterRepository.storeFilter(filterBody, roomFilter)
-        if (updated) {
-            val filterResponse = executeRequest(globalErrorReceiver) {
-                // TODO auto retry
-                filterAPI.uploadFilter(userId, filterBody)
-            }
-            filterRepository.storeFilterId(filterBody, filterResponse.filterId)
+    override suspend fun execute(params: SaveFilterTask.Params): String {
+        val filter = params.filter
+        val filterResponse = executeRequest(globalErrorReceiver) {
+            // TODO auto retry
+            filterAPI.uploadFilter(userId, filter)
         }
+
+        filterRepository.storeSyncFilter(
+                filter = filter,
+                filterId = filterResponse.filterId,
+                roomEventFilter = FilterFactory.createDefaultRoomFilter()
```

Comment:
It's weird to call `createDefaultRoomFilter`. Can you explain please?

Reply:
There is currently no difference between `ElementRoomFilter` and `DefaultRoomFilter` so basically there is no need to introduce any options here. It's also looks like we could utilise Filter.room.timeline filter for the purpose of `roomEventFilter`, which will mean that we don't need this separate filter at all. But currently no one can say for sure if we could/should do so or no, so for now I decided to do as little change to this area as possible, since the task was mostly about SyncFilter. I just wanted to remove referencing Element from the sdk, since there was no real difference between Default and Element room filters